File: math/1362_closest_divisors/1362_closest_divisors.go
package main

import (
	"fmt"
	"math"
)

func closestDivisors(num int) []int {
	v1 := helper(num + 1)
	v2 := helper(num + 2)
	if v1[1]-v1[0] < v2[1]-v2[0] {
		return v1
	} else {
		return v2
	}
}

func helper(num int) []int {
	root := int(math.Sqrt(float64(num)))
	for i := root; i >= 1; i-- {
		if num%i == 0 {
			return []int{i, num / i}
		}
	}
	return []int{}
}

func main() {
	num := 8
	res := closestDivisors(num)
	fmt.Println(res)
}

File: math/1362_closest_divisors/1362_closest_divisors.py

class Solution(object):
    def closestDivisors(self, num):
        for i in reversed(range(1, int((num + 2) ** 0.5) + 1)):
            if not (num +1) % i:
                return [i, (num+1)//i]
            if not (num+2) % i:
                return [i, (num+2)//i]


num = 8 
res = Solution().closestDivisors(num)
print(res)